[Merged by Bors] - Clarify why we only support HMS (#328)
stackabletech/hive-operator: docs/modules/hive/pages/index.adoc

= Stackable Operator for Apache Hive

This is an operator for Kubernetes that can manage https://hive.apache.org[Apache Hive] metastores.

> Currently it only supports running a Hive metastore service (HMS) that stores metadata about schemas and tables.

Reviewer: ```suggestion
Currently it only supports running a Hive metastore service (HMS) that stores metadata about schemas and tables.

Due to several reasons we do not feel that running Hive on Kubernetes makes much sense these days. The most obvious reason being that Hive requires Yarn as execution framework, which basically tries to do the same job as Kubernetes - i.e. assign resources. 
For this reason in the Stackable Data Platform we decided to provide [Trino](https://github.com/stackabletech/trino-operator) as query engine instead of Hive - which still uses the Hive Metastore, hence this operator exists.
```

> Currently it does only support running a Hive metastore service (HMS) that stores metadata about schemas and tables.

Reviewer 2: ```suggestion
Currently it only supports running a Hive metastore service (HMS) that stores metadata about schemas and tables.
```

There are multiple tools that can use the HMS:

* Hive server

> ** This this is the "original" tool using the HMS.

Reviewer: ```suggestion
** This is the "original" tool using the HMS.
```

** It offers a SQL endpoint, where you can submit HiveQL (similar to SQL) queries.
** It needs a execution engine, e.g. YARN or Spark.

> *** The reason it is currently not supported on the Stackable Platform is the complexity needed to operate YARN on kubernetes as well as the misconception of running a resource manager (YARN) on another resource manager (Kubernetes)

Reviewer 2: ```suggestion
*** This operator does not support running the Hive server because of  the complexity needed to operate YARN on Kubernetes. YARN is a resource manager which is not meant to be running on Kubernetes as Kubernetes already manages its own resources.
```

Reviewer 2: I didn't want the sentence to start with "The reason it is currently not supported"

If you feel my reformulation changes too much of the original meaning, please ignore :)

Author: Makes sense to me, thx!

> *** The reason it is currently not supported on the Stackable Platform is the complexity needed to operate YARN on kubernetes as well as the misconception of running a resource managed (YARN) on another resource managed (Kubernetes)

Reviewer 3: ```suggestion
*** The reason it is currently not supported on the Stackable Platform is the complexity needed to operate YARN on kubernetes as well as the misconception of running a resource manager (YARN) on another resource manager (Kubernetes)
```

*** We offer Trino as a (often times drop-in) replacement (see below)
* Trino
** Takes SQL queries and executes them against the tables, whose metadata are stored in HMS

> ** It should offer all the capabilities Hive offers including lot#s of additional functionality, such as connections to other datasources.

Reviewer 3: ```suggestion
** It should offer all the capabilities Hive offers including a lot of additional functionality, such as connections to other data sources.
```

= Stackable Operator for Apache Hive

This is an operator for Kubernetes that can manage https://hive.apache.org[Apache Hive] metastores.

Currently it only supports running a Hive metastore service (HMS) that stores metadata about schemas and tables.

> Due to several reasons we do not feel that running Hive on Kubernetes makes much sense these days. The most obvious reason being that Hive requires Yarn as execution framework, which basically tries to do the same job as Kubernetes - i.e. assign resources. 

Reviewer 4: ```suggestion
Due to several reasons we do not feel that running Hive on Kubernetes makes much sense these days. The most obvious reason being that Hive requires YARN as execution framework, which basically tries to do the same job as Kubernetes - i.e. assign resources. 
```

> For this reason in the Stackable Data Platform we decided to provide [Trino](https://github.com/stackabletech/trino-operator) as query engine instead of Hive - which still uses the Hive Metastore, hence this operator exists.

Reviewer 4: ```suggestion
For this reason we decided to provide [Trino](https://github.com/stackabletech/trino-operator) as query engine in the Stackable Data Platform instead of Hive - which still uses the Hive Metastore, hence this operator exists.
```

There are multiple tools that can use the HMS:

> * Hive server

Reviewer 4: ```suggestion
* HiveServer2
```

** This is the "original" tool using the HMS.

> ** It offers a SQL endpoint, where you can submit HiveQL (similar to SQL) queries.

Reviewer 4: ```suggestion
** It offers an endpoint, where you can submit HiveQL (similar to SQL) queries.
```
